srml/contract/src/exec.rs

```diff
@@ -106,8 +106,8 @@ pub trait Ext {
 	/// Returns a reference to the timestamp of the current block
 	fn now(&self) -> &MomentOf<Self::T>;
 
-	/// Returns a reference to the random seed for the current block
-	fn random_seed(&self) -> &SeedOf<Self::T>;
+	/// Returns a random number for the current block with the given subject.
+	fn random(&self, subject: &[u8]) -> SeedOf<Self::T>;
```

Comment:
Is the type alias name `SeedOf` still appropriate? Comment and docs refer to a "random number"

Reply:
SeedOf is defined this way `pub type SeedOf<T> = <T as system::Trait>::Hash;`
so user more likely want to use it as a seed than a full number generated maybe

Reply:
By this definition: https://en.wikipedia.org/wiki/Random_seed, isn't [this](https://github.com/paritytech/substrate/blob/318d2e0c9c81086513bb75b42223cbf080ed4993/srml/system/src/lib.rs#L589) the seed?

Reply:
Yes, as you pointed

> A seed is a vector used to initialize a pseudorandom number generator.

If you consider the pseudorandom number generator: the generator of the resulting hash then the seed is RandomMaterial and subject.
But I would think that ppl want to have pseudorandom number generator that generate more than just 32 u8 numbers.
In this case they would use the resulting hash as a seed of whatever pseudorandom generator they choose.

Though I'm not against renaming it, as it could be used as a random number and not as a seed.

Reply:
Ok cool I think the name is fine then. Thanks for clarifying!